WordPress/src/main/java/org/wordpress/android/networking/GravatarApi.java

```diff
@@ -76,17 +81,41 @@ public void run() {
                                 if (response.isSuccessful()) {
                                     gravatarUploadListener.onSuccess();
                                 } else {
+                                    Map<String, Object> properties = new HashMap<>();
+                                    properties.put("network_response_code", response.code());
+
+                                    String responseBody;
+                                    try {
+                                        responseBody = response.body().string();
+                                    } catch (IOException e) {
+                                        responseBody = "null";
+                                    }
+                                    properties.put("network_response_body", responseBody);
+
+                                    AnalyticsTracker.track(AnalyticsTracker.Stat.ME_GRAVATAR_UPLOAD_UNSUCCESSFUL,
+                                            properties);
+                                    AppLog.w(AppLog.T.API, "Network call unsuccessful trying to upload Gravatar: " +
+                                            responseBody);
                                     gravatarUploadListener.onError();
                                 }
                             }
                         });
                     }
 
                     @Override
-                    public void onFailure(okhttp3.Call call, IOException e) {
+                    public void onFailure(okhttp3.Call call, final IOException e) {
                         new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
It is probably not making any real difference in this case (we do bump Analytics in the main thread throughout the app anyway) but now that you mention it, it will be a bit cleaner code to do it before the Handler post. I'll make the change ;)